### PR TITLE
[Merged by Bors] - Extract the display logic from the listener table, make it sortable

### LINKED ISSUE #3: Create data table component
Ideally, this should (in the future?) let people sort and filter the data shown for a view, in a reusable way.

### PR DESCRIPTION
**Pull request by nightkr**
# Description

Fixes #3

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
# Author
- [ ] Changes are OpenShift compatible
- [ ] CRD changes approved
- [ ] Helm chart can be installed and deployed operator works
- [ ] Integration tests passed (for non trivial changes)
```

```[tasklist]
# Reviewer
- [ ] Code contains useful comments
- [ ] (Integration-)Test cases added
- [ ] Documentation added or updated
- [ ] Changelog updated
- [ ] Cargo.toml only contains references to git tags (not specific commits or branches)
```

```[tasklist]
# Acceptance
- [ ] Feature Tracker has been updated
- [ ] Proper release label has been added
```

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: web/src/components/datatable.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  const sortedItems = createMemo(() => {
+    const currentSortComparator = sortComparator();
+    if (currentSortComparator == undefined) {
+      return props.items;
+    } else {
+      const items = [...props.items];
+      items.sort(currentSortComparator);
+      return items;
+    }
+  });
```
Comment (Techassi) [inline web/src/components/datatable.tsx:25]: Add a newline between elements. This should help readability.

```suggestion
export interface DataTableColumn<T> {
  label: string;
  get: (x: T) => JSX.Element;
  sortBy?: (x: T) => string;
}

export interface DataTableProps<T> {
  columns: DataTableColumn<T>[];
  items: T[];
}

export function DataTable<T>(props: DataTableProps<T>): JSX.Element {
  const [sortComparator, setSortComparator] = createSignal<
    ((x: T, y: T) => number) | undefined
  >();
  
  const sortedItems = createMemo(() => {
    const currentSortComparator = sortComparator();
    if (currentSortComparator == undefined) {
      return props.items;
    } else {
      const items = [...props.items];
      items.sort(currentSortComparator);
      return items;
    }
  });
```

File: web/src/components/datatable.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+                  <Show when={col.sortBy} fallback={col.label}>
+                    <a
+                      href=""
+                      class="text-gray-400"
+                      onClick={(event) => {
+                        event.preventDefault();
+                        setSortComparator(() =>
+                          col.sortBy ? keyComparator(col.sortBy) : undefined,
+                        );
+                      }}
```
Comment (Techassi) [inline web/src/components/datatable.tsx:43]: Might be worth to move this into a function defined above the TSX.

**Review by Techassi — COMMENTED** [review]
Minor changes. Other than that lgtm.

File: web/src/components/datatable.tsx (hunk 2 above)
Comment (nightkr) [reply web/src/components/datatable.tsx]: 35f9d99e08e6cca050ebf07c5532fb7b80f1553f

**Review by Techassi — APPROVED** [review]

**Pull request closed without merging**